util/time.go

```diff
@@ -130,3 +130,14 @@ func GetUTCHour(date time.Time) time.Time {
 	hour := date.Hour()
 	return time.Date(year, month, day, hour, 0, 0, 0, time.UTC)
 }
+
+func GetUTCSecond(date time.Time) time.Time {
```

Comment:
I would prefer us not to add a new time function. `util.RoundPartOfMinute(n).Format(tsFormat)` creates the time using time.Now() which is what your calls to `GetUTCSecond` are taking in anyway, unless you really specifically can't have the second be rounded.